### PR TITLE
feat: various relatively minor feats and fixes

### PR DESCRIPTION
**Pull request by edkerk**
### Main improvements in this PR:
- feat:
  - wrapper IO functions `loadEcModel`, `saveEcModel`, `loadConventionalGEM`, that can take modelAdapter info as input, to simplify loading and saving of models. It's just a wrapper for `readYAMLmodel`, `writeYAMLmodel` and `importModel`, so these can still be used as well.
  - `getKcatAcrossIsoenzymes` assigns mean kcat to kcat-less reactions that have isoenzymes with kcat values.
  - `loadDatabases` can output alternative gene annnotations instead of KEGG gene IDs. `kegg.tsv` therefore has an additional column, to ensure that kegg gene ID is still present. Also, `uniprot.tsv` has been slightly shrunk by removing unused gene names annotation.
  - `applyCustomKcats` also allows providing only the original reaction ID (without the _EXP_ specification) and no protein IDs. In that case, it will apply the same kcat to all isoenzymes.

- fix:
  - `mergeDLKcatAndFuzzyKcats` changed default parameters, to solve #205. Also documentation slightly extended to exemplify the meaning of the parameters.

- chore:
  - remove deprecated files and functions

**I hereby confirm that I have:**

- [ ] Tested my code with [all requirements](https://github.com/SysBioChalmers/GECKO) for running GECKO
- [x] Selected `devel3` as a target branch (top left drop-down menu)

### CODE REVIEW

File: src/geckomat/change_model/getKcatAcrossIsoenzymes.m [hunk 1]
```diff
@@ -0,0 +1,38 @@
+function model = getKcatAcrossIsoenzymes(model)
+% getKcatAcrossIsoenzymes
```
Comment (johan-gson) [inline src/geckomat/change_model/getKcatAcrossIsoenzymes.m:2]: I guess this is only relevant for full ec models? May be good to write in the text and check in the function.

Comment (johan-gson) [reply src/geckomat/change_model/getKcatAcrossIsoenzymes.m:2]: Will you add a test case for this function? It should be relatively easy - you could copy parts of tc0011 and just zero the kcat for reaction R2_EXP_2 or something like that. And then make sure it gets filled in from R2_EXP_1. Make sure it works also for REV.

**Review by ae-tafur — APPROVED** [review]
Looks good! 

Should we include here as deprecated `utilities/changeMedia_yeast.m`, or move this to `userData/ecYeastGEM/code` ?

File: src/geckomat/change_model/getKcatAcrossIsoenzymes.m (hunk 1 above)
Comment (edkerk) [reply src/geckomat/change_model/getKcatAcrossIsoenzymes.m:2]: Or would you want something similar for light models? Then the smallest isoenzyme will always be selected, while if that isoenzyme would not have had a kcat value it would never be active? (I assume that light ignores isoenzymes with 0 kcat value?)

**Comment by edkerk** [comment]
> Looks good!
> 
> Should we include here as deprecated `utilities/changeMedia_yeast.m`, or move this to `userData/ecYeastGEM/code` ?

Good suggestion, will have to be refactored later.

File: src/geckomat/utilities/saveEcModel.m [hunk 2]
```diff
@@ -0,0 +1,69 @@
+function saveECmodel(model,modelAdapter,format,filename)
```
Comment (johan-gson) [inline src/geckomat/utilities/saveEcModel.m:1]: Could you update tc0009 to test these three functions, i.e., load a conventional GEM, save it as ec, load it, and compare? Should be easy. Ideally the model should go through the processing chain so we test that metSmiles etc. are saved.

**Review by johan-gson — APPROVED** [review]
I have some comments, mostly about testing. At this point, I think it would be good to add test cases for the new functions we add - we need to test them at some point anyway.

**Comment by johan-gson** [comment]
Another comment I forgot, about Human-GEM. I got the feeling that the model is stored in GECKO, is that so? Human-GEM still needs to be installed, we use gene conversions etc. from that installation - isn't it better to load the model from it's real location?

File: src/geckomat/change_model/getKcatAcrossIsoenzymes.m (hunk 1 above)
Comment (johan-gson) [reply src/geckomat/change_model/getKcatAcrossIsoenzymes.m:2]: Yeah, they are ignored, but then it would be the mean of the other possibilities, and it would not be the smallest, and not be used :). I don't think we need one here - it is only if you constrain individual enzymes that this makes sense unless I misunderstood something?

Comment (edkerk) [reply src/geckomat/change_model/getKcatAcrossIsoenzymes.m:2]: Right, indeed, I'll then just let it throw an error if applied to geckoLight model.

**Comment by edkerk** [comment]
> Another comment I forgot, about Human-GEM. I got the feeling that the model is stored in GECKO, is that so? Human-GEM still needs to be installed, we use gene conversions etc. from that installation - isn't it better to load the model from it's real location?

Probably. The model is currently not in GECKO. I've changed the adapter to refer to the Human-GEM location, similar as is done for getting Uniprot compatible genes
